drgn/helpers/linux/mm.py

```diff
@@ -291,3 +294,26 @@ def environ(task: Object) -> List[bytes]:
     env_start = mm.env_start.value_()
     env_end = mm.env_end.value_()
     return access_remote_vm(mm, env_start, env_end - env_start).split(b"\0")[:-1]
+
+
+def find_slab(prog: Program, slab_name: str) -> Optional[Object]:
+    """
+    Return the slab object corresponding to slab_name
+
+    :param slab_name: name of the slab
+    :return: ``struct kmem_cache``
+    """
+    slab_caches_addr = prog["slab_caches"].address_of_().read_()
+    for s in list_for_each_entry("struct kmem_cache", slab_caches_addr, "list"):
+        if s.name.string_().decode("utf-8") == slab_name:
+            return s
+    return None
+
+
+def print_slabs(prog: Program) -> None:
+    """
+    Print the slab name and the value of each ``struct kmem_cache *`` for all slabs.
+    """
+    slab_caches_addr = prog["slab_caches"].address_of_().read_()
+    for s in list_for_each_entry("struct kmem_cache", slab_caches_addr, "list"):
+        print(f"{s.name.string_().decode()} ({s.type_.type_name()})0x{s.value_():x}")
```

Comment:
You should use `escape_ascii_string()` from `drgn.helpers` instead of `.decode()` to protect against a garbage name here.

```suggestion
        name = escape_ascii_string(s.name.string_(), escape_backslash=True)
        print(f"{name} ({s.type_.type_name()})0x{s.value_():x}")
```